blog/2025-08-00-uppy-5.0.mdx

```diff
@@ -0,0 +1,772 @@
+import { BrowserWindow } from '/src/components/BrowserWindow';
+
+The dog didn't eat _our_ homework, so we're here to show you what we've been up
+to over the past year. We are proud to announce Uppy 5.0! Our faithful
+file-uploading companion has been attending all the training courses and is now
+equipped with the most requested feature of all time: truly composable, headless
+UI components. For years, developers have found themselves stuck between
+Dashboard’s kitchen-sink approach and the bare-bones drag-drop plugin, with
+little room to build something that fits just right.
+
+This major release brings headless components and hooks to address those needs.
+Everything has been backported to 4.0 as well, so it's easy to get going without
+the need to upgrade.
+
+With Uppy 5.0 , you can finally build the exact upload experience your users
+deserve – modular, accessible, and tailored to your needs. Whether you need a
+simple file picker or a complex multi-source interface, Uppy’s got the building
+blocks to make it happen.
+
+{/* truncate */}
+
+## Building your UI: the need for something new
+
+### One size fits no one
+
+Until now, developers starting with Uppy were forced to choose between either
+the full-featured, bundle-size-heavy, and non-customizable `dashboard` or the
+overly barebones `drag-drop`.
+
+After years of talking to developers on GitHub and the community forum, and
+feedback from Transloadit customers, the reality seems to be that the needs of
+the majority of people fall somewhere _in between_ `dashboard` and `drag-drop`.
+Countless issues have been posted about users wanting mostly X, but doing Y
+differently for their use case.
+
+`@uppy/dashboard` has tried to accommodate some of these requests over the
+years, which introduced overly specific “toggles”, such as
+`showLinkToFileUploadResult`, `showProgressDetails`, `hideUploadButton`,
+`hideRetryButton`, `hidePauseResumeButton`, `hideCancelButton`,
+`hideProgressAfterFinish`, `showRemoveButtonAfterComplete`, `disableStatusBar`,
+`disableInformer`, and `disableThumbnailGenerator`.
+
+Continuing down this path is not maintainable, nor will we ever reach a set of
+toggles at this level of granularity to support a wide range of use cases and
+unique requirements.
+
+### Fake promises of modular UI components
+
+We built `status-bar`, `progress-bar`, `image-editor`, `thumbnail-generator`,
+`informer`, and `provider-views` as separate plugins. At first glance, these
+look to be standalone components, but they are tightly coupled to `dashboard`.
+It’s not impossible to use them separately, but there is little flexibility.
+
+### Modern expectations
+
+Since Uppy’s conception ten years ago, the front-end landscape has changed
+significantly. Uppy is even built with a “vanilla-first" approach, because that
+was still popular at the time over front-end frameworks.
+
+**These days, developers have high expectations for choosing a library**. In a
```

Comment:
I find this sentence slightly unclear, mainly the word 'for'. What do you mean to say? That developers have high expectations _when it comes to_ choosing a library? Or that they now _expect to_ choose a library?